### PR TITLE
perf: mark warming

### PR DESCRIPTION
**Pull request by arthyn**
This change makes a "mark warmer" file which ensures that particular conversions are pre-compiled and cached so that we aren't doing the compilation on every request. This gives us a tremendous boost to perf. Thanks to @pkova and @philipcmonk

### CODE REVIEW

File: talk/app/talk-ui.hoon [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /-  u=ui, g=groups, c=chat
 /+  default-agent, dbug, verb, vita-client
+/%  init  %ui-init
```
Comment (joemfb) [inline talk/app/talk-ui.hoon:3]: should this be `/$  init  %ui-init  %json`?

**Review by philipcmonk — APPROVED** [review]
Looks good

File: desk/app/chat.hoon [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
 /+  default-agent, verb-lib=verb, dbug
-/+  chat-json
 /+  pac=dm
 /+  ch=chat-hark
 /+  gra=graph-store
 /+  epos-lib=saga
 /+  wood-lib=wood
 /+  mig=chat-graph
+::  performance, keep warm
+/+  chat-json
```
Comment (joemfb) [inline desk/app/chat.hoon:14]: Is this (and all the other `*-json` lib imports in the %groups desk) still necessary with `lib/mark-warmer` being imported by `:groups-ui`?

**Review by joemfb — COMMENTED** [review]
A couple questions, otherwise looks good.

File: desk/app/chat.hoon (hunk 2 above)
Comment (arthyn) [reply desk/app/chat.hoon:14]: good question I'm not sure

File: talk/app/talk-ui.hoon (hunk 1 above)
Comment (arthyn) [reply talk/app/talk-ui.hoon:3]: yeah you're right

```suggestion
/$  init  %ui-init  %json
```

**Pull request merged**